File: app/src/main/java/com/example/android/volleyballscoreboard/SettingsActivity.java
package com.example.android.volleyballscoreboard;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import com.example.android.volleyballscoreboard.databinding.ActivitySettingsBinding;

/**
 * Created by Oya on 22-02-18.
 */

public class SettingsActivity extends AppCompatActivity {

    private SettingsViewModel mViewModel;
    private ActivitySettingsBinding binding;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings);
        mViewModel = ViewModelProviders.of(this).get(SettingsViewModel.class);
        binding.setViewModel(mViewModel);

        //Initialize spinners
        initializeSpinner(R.array.numberOfSets, binding.spin1);
        initializeSpinner(R.array.setFinishingScore, binding.spin2);
        initializeSpinner(R.array.tieBreakerScore, binding.spin3);
    }

    private void initializeSpinner(int arrayId, Spinner spinner) {
        ArrayAdapter<CharSequence> spinAdapter1 = ArrayAdapter.createFromResource(this,
                arrayId, R.layout.spinner_item);
        spinAdapter1.setDropDownViewResource(R.layout.spinner_dropdown);
        spinner.setAdapter(spinAdapter1);
    }

    public void startGame(View view) {
        Intent userChoices = new Intent(SettingsActivity.this, MainActivity.class);
        userChoices.putExtra(Constants.TEAM_NAME_ORANGE, mViewModel.getOrangeTeamName());
        userChoices.putExtra(Constants.TEAM_NAME_BLUE, mViewModel.getBlueTeamName());
        userChoices.putExtra(Constants.STARTING_TEAM, mViewModel.getStarterTeamId());
        userChoices.putExtra(Constants.NUMBER_OF_TOTAL_SETS, mViewModel.getNumbersOfTotalSets());
        userChoices.putExtra(Constants.SET_FINISHING_SCORE, mViewModel.getSetFinishingScore());
        userChoices.putExtra(Constants.TIE_BREAKER_SCORE, mViewModel.getTieBreakerScore());
        startActivity(userChoices);
    }

    public void flipCoin(View view) {
        double chance = Math.random();
        chance *= 2;
        if (chance < 1) { //heads
            binding.coin.setImageResource(R.drawable.head);
        } else { //tails
            binding.coin.setImageResource(R.drawable.tails);
        }
    }
}

File: app/src/main/java/com/example/android/volleyballscoreboard/SettingsActivity.kt
package com.example.android.volleyballscoreboard

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProviders
import com.example.android.volleyballscoreboard.databinding.ActivitySettingsBinding

/**
 * Created by Oya on 22-02-18.
 */

class SettingsActivity : AppCompatActivity() {

    private val mViewModel by lazy {
        ViewModelProviders.of(this).get(SettingsViewModel::class.java)
    }
    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings)
        binding.viewModel = mViewModel

        //Initialize spinners
        initializeSpinner(R.array.numberOfSets, binding.spin1)
        initializeSpinner(R.array.setFinishingScore, binding.spin2)
        initializeSpinner(R.array.tieBreakerScore, binding.spin3)
    }

    private fun initializeSpinner(arrayId: Int, spinner: Spinner) {
        val spinAdapter = ArrayAdapter.createFromResource(this,
                arrayId, R.layout.spinner_item)
        spinAdapter.setDropDownViewResource(R.layout.spinner_dropdown)
        spinner.adapter = spinAdapter
    }

    fun startGame(view: View) {
        val userChoices = Intent(this@SettingsActivity, MainActivity::class.java)
        with(mViewModel) {
            userChoices.putExtra(KEY_TEAM_NAME_ORANGE, orangeTeamName)
            userChoices.putExtra(KEY_TEAM_NAME_BLUE, blueTeamName)
            userChoices.putExtra(KEY_STARTING_TEAM, starterTeamId)
            userChoices.putExtra(KEY_NUMBER_OF_TOTAL_SETS, numbersOfTotalSets)
            userChoices.putExtra(KEY_SET_FINISHING_SCORE, setFinishingScore)
            userChoices.putExtra(KEY_TIE_BREAKER_SCORE, tieBreakerScore)
        }
        startActivity(userChoices)
    }

    fun flipCoin(view: View) {
        var chance = Math.random()
        chance *= 2.0
        binding.coin.setImageResource(if (chance < 1) R.drawable.head else R.drawable.tails)
    }
}
